MovieQuiz/Presentation/MovieQuizPresenter.swift

```diff
@@ -0,0 +1,127 @@
+    private var viewController: MovieQuizViewControllerProtocol?
+    
+    private var correctAnswers: Int = 0 //
+    private var questionFactory: QuestionFactoryProtocol? = nil//
+    var alertPresenter: AlertPresenterProtocol? = nil//
+    private let statisticService: StatisticService? //
+    
+    init(viewController: MovieQuizViewControllerProtocol) {
+        self.viewController = viewController
+        
+        statisticService = StatisticServiceImplementation()
+        questionFactory = QuestionFactory(moviesLoader: MoviesLoader(), delegate: self)
+        questionFactory?.loadData()
+        viewController.showLoadingIndicator()
+    }
+    // MARK: - QuestionFactoryDelegate
+    func didLoadDataFromServer() {
+        viewController?.hideLoadingIndicator()
+        questionFactory?.requestNextQuestion()
+    }
+    
+    func didFailToLoadData(with error: Error) {
+        let message = error.localizedDescription
+        viewController?.showNetworkError(message: message)
+    }
+    
+    func didReceiveNextQuestion(question: QuizQuestion?) {
+        guard let question = question else {
+            return
+        }
+        
+        currentQuestion = question
+        let viewModel = convert(model: question)
+        DispatchQueue.main.async { [weak self] in
+            self?.viewController?.show(quiz: viewModel)
+        }
+    }
+    
+    func isLastQuestion() -> Bool {
+        currentQuestionIndex == questionsAmount - 1
+    }
+    
+    func restartGame() {
+        currentQuestionIndex = 0
+        correctAnswers = 0
+        questionFactory?.requestNextQuestion()
+    }
+    
+    func switchToNextQuestion() {
+        currentQuestionIndex += 1
+    }
+    
+    func didAnswer(isCorrectAnswer: Bool) {
+        if isCorrectAnswer {
+            correctAnswers += 1
+        }
+    }
+    
+    func convert(model: QuizQuestion) -> QuizStepViewModel {
+        return QuizStepViewModel(
+            image: UIImage(data: model.image) ?? UIImage(),
+            question: model.text,
+            questionNumber: "\(currentQuestionIndex + 1)/\(questionsAmount)")
+    }
+    
+    func yesButtonClicked() {
+        didAnswer(isYes: true)
+    }
+    func noButtonClicked() {
+        didAnswer(isYes: false)
+    }
+    private func didAnswer(isYes: Bool) {
+        guard let currentQuestion = currentQuestion else {
+            return
+        }
+        let givenAnswer = isYes
+        proceedWithAnswer(isCorrect: givenAnswer == currentQuestion.correctAnswer)
+    }
+    
+    func proceedToNextQuestionOrResults() {
+        
+        if self.isLastQuestion() {
+            statisticService?.store(correct: correctAnswers, total: questionsAmount)
+            guard
+                let bestGame = statisticService?.bestGame,
+                let gamesCount = statisticService?.gamesCount,
+                let totalAccuracy = statisticService?.totalAccuracy
+            else {
+                return
+            }
```

Comment:
если разыменовать здесь statisticService, то свойства можно будет доставать без разыменовывания

Сэкономишь строчки
```suggestion
            guard let statisticService else { return }
```

Reply:
если заменить на предложенную Вами строку, то тогда ниже выскакивают ошибки: 
Cannot find 'bestGame' in scope
Cannot find 'bestGame' in scope
Cannot find 'bestGame' in scope

Reply:
@Stanny-Starczewski конечно, указанного мною изменения не достаточно будет. Это лишь пример, где можно разыменовать один раз, вместо трех, так мы стираем три введенные переменные, вместо них надо будет обратиться к statisticService чтобы получить необходимые значения. Попробуй еще раз